src/app/persistence/String.h

```diff
@@ -0,0 +1,122 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/persistence/AttributePersistenceProvider.h>
+#include <app/persistence/PascalString.h>
+#include <lib/support/Span.h>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace chip::app::Storage {
+
+namespace Internal {
+
+/// Represents a string that can be read/written to storage.
+///
+/// This is currently implemented for SHORT strings only (< 255 bytes)
+/// that are NOT nullable
+class ShortString
+{
+public:
+    /// Use the input buffer as a short pascal string.
+    ///
+    /// The input buffer is assumed to have:
+    ///   - size for a PascalString (i.e. generally pascal string length + prefix == length + 1 given we use
+    ///     short pascal strings here)
+    ///   - an extra +1 bytes for a null terminator so that c_str works.
+    ///
+    /// This means that buffer_size should be `N+2` where N is the expected maximum string length.
+    ///
+    /// This class is considered Internal, use chip::app::Storage::String<MAX_LENGTH> in application code.
+    ShortString(char * buffer, size_t buffer_size) : mBuffer(buffer), mPascalSize(static_cast<uint8_t>(buffer_size - 1))
+    {
+        // for a buffer to be usable we need 1 byte for size, 1 byte for content and 1 byte for null terminator.
+        // Strings without any size make no sense.
+        VerifyOrDie(buffer_size >= 3);
+        VerifyOrDie(buffer_size < 256);
+    }
+
+    /// Returns the content as a character span
+    CharSpan Content() const { return AsPascal().Content(); }
+
+    /// Returns the content as a null terminated string
+    const char * c_str() const { return Content().data(); /* ALWAYS null terminated*/ }
+
+    /// sets the internal value of the string to the given value.
+    /// If the set fails, the value is set to empty string and false is returned.
+    bool SetContent(CharSpan value);
+
+    friend class ShortStringIO;
+
+private:
+    char * mBuffer;
+    uint8_t mPascalSize;
+
+    ShortPascalString AsPascal() { return { mBuffer, mPascalSize }; }
+    ShortConstPascalString AsPascal() const { return { mBuffer, mPascalSize }; }
+
+    /// Places a null terminator after the pascal string content, so that c_str() works
+    void NullTerminate() { mBuffer[AsPascal().ContentWithLenPrefix().size()] = 0; }
+};
+
+/// Provides internal access and processing to a ShortString class.
+///
+/// Meant for internal use only, this is NOT public API outside the SDK code itself.
+class ShortStringIO
+{
+public:
+    ShortStringIO(ShortString & value) : mValue(value) {}
+    MutableByteSpan ReadBuffer()
+    {
+        return { reinterpret_cast<uint8_t *>(mValue.mBuffer), static_cast<size_t>(mValue.mPascalSize) };
```

Comment:
This is basically the same as `mValue.AsPascal().RawFullBuffer()`?  If there's a reason to do it "by hand" instead of doing that, please document that reason (codesize?  Clearer what's really going on? Something else?

Note that we do use `AsPascal()` below in the similar setup in `ContentWithPrefix`